src/service.js

```diff
@@ -771,8 +771,12 @@ class Service {
 	 * @param {String|Number?} version
 	 */
 	static getVersionedFullName(name, version) {
-		if (version != null)
-			return (typeof(version) == "number" ? "v" + version : version) + "." + name;
+		if (version != null) {
+			if (typeof(version) === 'number' || !isNaN(parseInt(version))) {
```

Comment:
This is a breaking change if `version = "123abc"`.

Reply:
Good call, I just updated this to use the Number constructor instead and I added a test case to validate that it works the way we would expect. 

Thanks!

Reply:
It's technically also breaking for anyone who is using `version = "5"` intentionally... I understand your pain here though.

Reply:
That's also true, I would think this would be a "good" breaking change from my perspective. Let me update the PR to flag that this is breaking and let @icebob think on it 🙂